source/core/module_coreHelpers.js

```diff
@@ -56,15 +56,18 @@
             fpSync = fn;
         };
 
-        Module.coreHelpers.utf8ArrayToStringWithNull = function (u8Array, idxIn) {
+        // Takes Vireo Strings that are UTF-8 encoded strings with known length and returns a JS string
+        Module.coreHelpers.sizedUtf8ArrayToJSString = function (u8Array, startIndex, length) {
             /* eslint-disable no-continue, no-plusplus, no-bitwise */
             var u0, u1, u2, u3, u4, u5;
-            var idx = idxIn;
+            var idx = startIndex;
+            var lastVisitedIndex = startIndex + length;
```

Comment:
The function takes a UTF8Array of bytes, a start index, and a length (number of utf8 bytes). We are not passing in a string, but instead a UT8Array of bytes and it returns a JS string.

One caller happens to use the whole buffer so it passes a startindex of 0 and the length of the the whole buffer (vireo.loader.js)
Another caller actually has a starting index and length to extract a subset of the buffer (see module_eggShell.js)

Does seeing both usages make it clearer?
Would startIndex (inclusive) and endIndex (exclusive) make more sense than startIndex + length for the api?